### PR TITLE
Add auto-reset document

### PR DESCRIPTION
**Pull request by longquanzheng**

### CODE REVIEW

File: docs/08_cli.md [hunk 1]
```diff
@@ -201,3 +201,24 @@ To reset multiple workflows, you can use batch reset command:
 ```
 ./cadence workflow reset-batch --input_file <file_of_workflows_to_reset> --reset_type <reset_type> --reason "some_reason"
 ```
+#### Recovery from bad deployment -- auto-reset workflow
+If a bad deployment let a workflow run into a wrong state, you might want to reset the workflow to the point that the bad deployment started to run. But usually it is not easy to find out all the workflows impacted, and every reset point for each workflow. In this case, auto-reset will automatically reset all the workflow by the bad deployment given a identfier.
```
Comment (purplesusan) [inline docs/08_cli.md]: let -> lets
all the workflow -> all the workflows
a identifier -> an identifier

File: docs/08_cli.md [hunk 2]
```diff
@@ -201,3 +201,24 @@ To reset multiple workflows, you can use batch reset command:
 ```
 ./cadence workflow reset-batch --input_file <file_of_workflows_to_reset> --reset_type <reset_type> --reason "some_reason"
 ```
+#### Recovery from bad deployment -- auto-reset workflow
+If a bad deployment let a workflow run into a wrong state, you might want to reset the workflow to the point that the bad deployment started to run. But usually it is not easy to find out all the workflows impacted, and every reset point for each workflow. In this case, auto-reset will automatically reset all the workflow by the bad deployment given a identfier. 
+
+First of all, let's familar with some concepts. Each deployment will have a identifier, we call it "**Binary Checksum**" as it is usually generated by the md5sum of a binary file. For a workflow, each binary checksum will be associated with an **auto-reset point**, which contains a **runID**, an **eventID**, and the **created_time** that binary/deployment made the first decision for the workflow.
```
Comment (purplesusan) [inline docs/08_cli.md]: let's familiar -> let's get familiar

File: docs/08_cli.md [hunk 3]
```diff
@@ -201,3 +201,24 @@ To reset multiple workflows, you can use batch reset command:
...
+
+To find out which **binary checksum** of bad deployment to reset, you should be aware of at least one workflow running into a bad state. Use describe command with **--reset_points_only** option to show all the reset points:
```
Comment (purplesusan) [inline docs/08_cli.md]: of bad deployment -> of the bad deployment
Use describe command -> Use the describe command

File: docs/08_cli.md [hunk 4]
```diff
@@ -201,3 +201,24 @@ To reset multiple workflows, you can use batch reset command:
...
++----------------------------------+--------------------------------+--------------------------------------+---------+
+|         BINARY CHECKSUM          |          CREATE TIME           |                RUNID                 | EVENTID |
++----------------------------------+--------------------------------+--------------------------------------+---------+
+| c84c5afa552613a83294793f4e664a7f | 2019-05-24 10:01:00.398455019  | 2dd29ab7-2dd8-4668-83e0-89cae261cfb1 |       4 |
+| aae748fdc557a3f873adbe1dd066713f | 2019-05-24 11:01:00.067691445  | d42d21b8-2adb-4313-b069-3837d44d6ce6 |       4 |
+...
+...
+```
+
+Then use this command to tell Cadence auto-reset all workflows impacted by the bad deployment. The bad binary checksum needs to store in domain info.
```
Comment (purplesusan) [inline docs/08_cli.md]: Cadence auto-reset -> Cadence to auto-reset

I'm not sure what this means: The bad binary checksum needs to store in domain info. What is being stored and where?

Comment (longquanzheng) [reply docs/08_cli.md]: Oh if you look at the command, it is updating the domain info. I am trying explain why. But maybe it is not necessary if it leads more confusion.

Comment (longquanzheng) [reply docs/08_cli.md]: I changed it a little bit hope it makes more sense.

Comment (purplesusan) [reply docs/08_cli.md]: Your update does make more sense :) Thanks!

File: docs/08_cli.md [hunk 5]
```diff
@@ -201,3 +201,24 @@ To reset multiple workflows, you can use batch reset command:
 ```
 ./cadence workflow reset-batch --input_file <file_of_workflows_to_reset> --reset_type <reset_type> --reason "some_reason"
 ```
+#### Recovery from bad deployment -- auto-reset workflow
+If a bad deployment lets a workflow run into a wrong state, you might want to reset the workflow to the point that the bad deployment started to run. But usually it is not easy to find out all the workflows impacted, and every reset point for each workflow. In this case, auto-reset will automatically reset all the workflows given a bad deployment identfier.
```
Comment (dmetzgar) [inline docs/08_cli.md]: Fix spelling on "identfier"

File: docs/08_cli.md [hunk 6]
```diff
@@ -201,3 +201,24 @@ To reset multiple workflows, you can use batch reset command:
 ```
 ./cadence workflow reset-batch --input_file <file_of_workflows_to_reset> --reset_type <reset_type> --reason "some_reason"
 ```
+#### Recovery from bad deployment -- auto-reset workflow
+If a bad deployment lets a workflow run into a wrong state, you might want to reset the workflow to the point that the bad deployment started to run. But usually it is not easy to find out all the workflows impacted, and every reset point for each workflow. In this case, auto-reset will automatically reset all the workflows given a bad deployment identfier. 
+
+First of all, let's get familar with some concepts. Each deployment will have an identifier, we call it "**Binary Checksum**" as it is usually generated by the md5sum of a binary file. For a workflow, each binary checksum will be associated with an **auto-reset point**, which contains a **runID**, an **eventID**, and the **created_time** that binary/deployment made the first decision for the workflow.
```
Comment (dmetzgar) [inline docs/08_cli.md]: Remove "First of all, let's get familiar with some concepts." The sentence has no value.

File: docs/08_cli.md [hunk 7]
```diff
@@ -201,3 +201,24 @@ To reset multiple workflows, you can use batch reset command:
...
++----------------------------------+--------------------------------+--------------------------------------+---------+
+|         BINARY CHECKSUM          |          CREATE TIME           |                RUNID                 | EVENTID |
++----------------------------------+--------------------------------+--------------------------------------+---------+
+| c84c5afa552613a83294793f4e664a7f | 2019-05-24 10:01:00.398455019  | 2dd29ab7-2dd8-4668-83e0-89cae261cfb1 |       4 |
+| aae748fdc557a3f873adbe1dd066713f | 2019-05-24 11:01:00.067691445  | d42d21b8-2adb-4313-b069-3837d44d6ce6 |       4 |
+...
+...
+```
+
+Then use this command to tell Cadence to auto-reset all workflows impacted by the bad deployment. Essentailly the command will store the bad binary checksum into domain info and trigger a process to reset all your workflows.
```
Comment (dmetzgar) [inline docs/08_cli.md]: Remove the word "Essentailly" as it's misspelled and it doesn't contribute anything to the sentence.

**Pull request merged**